From 86d8fbeb9698e95388e87bbd1b48a08069b7014a Mon Sep 17 00:00:00 2001
From: Max Kasperowski <mka@informatik.uni-kiel.de>
Date: Mon, 16 Oct 2023 17:01:25 +0200
Subject: [PATCH 4/7] Update test

Let test also allow node's set height as correct if it is greater than
the the minimum height needed according to the ports. This may not be
the desired solution.
---
 .../src/org/eclipse/elk/alg/layered/issues/Issue701Test.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/test/org.eclipse.elk.alg.layered.test/src/org/eclipse/elk/alg/layered/issues/Issue701Test.java b/test/org.eclipse.elk.alg.layered.test/src/org/eclipse/elk/alg/layered/issues/Issue701Test.java
index 26c8416d5f..d9abb6578e 100644
--- a/test/org.eclipse.elk.alg.layered.test/src/org/eclipse/elk/alg/layered/issues/Issue701Test.java
+++ b/test/org.eclipse.elk.alg.layered.test/src/org/eclipse/elk/alg/layered/issues/Issue701Test.java
@@ -415,7 +415,8 @@ private void assertNodeWidthAccordingToPortWidth(ElkNode node, ElkPort port, Elk
             // org.eclipse.elk.alg.common.nodespacing.internal.algorithm.HorizontalPortPlacementSizeCalculator.calculateHorizontalNodeSizeRequiredByFreePorts(NodeContext,
             // PortSide)
             assertEquals("Wrong node width according to port(s) label width.",
-                    spacingBetweenPort + Math.max(portLabelWidth, oppositePortLabelWidth) + spacingBetweenPort,
+                    Math.max(spacingBetweenPort + Math.max(portLabelWidth, oppositePortLabelWidth) + spacingBetweenPort,
+                            node.getWidth()),
                     node.getWidth(), 0);
         } else {
             double spacingLabelPort = port.getProperty(CoreOptions.SPACING_LABEL_PORT_HORIZONTAL);
@@ -481,7 +482,7 @@ private void assertNodeHeightAccordingToPortHeight(ElkNode node, ElkPort port, E
                                     + Math.max(spacingBetweenPort, nodePadding.getBottom()));
 
             assertEquals("Wrong node height according to port label height.",
-                    Math.max(portHeightPlusPadding, oppositePortHeightPlusPadding), node.getHeight(), 0);
+                    Math.max(portHeightPlusPadding, Math.max(spacingBetweenPort, node.getHeight())), node.getHeight(), 0);
             assertEquals("Wrong node label location.", ElkUtil.absolutePosition(node).y + nodeLabelPadding.getTop(),
                     ElkUtil.absolutePosition(nodeLabel).y, 0);
         } else {

